spec/plaindate.html

```diff
@@ -756,8 +756,8 @@ <h1>ToTemporalDate ( _item_ [ , _options_ ] )</h1>
           1. Let _fields_ be ? PrepareTemporalFields(_item_, _fieldNames_, «»).
           1. Return ? CalendarDateFromFields(_calendar_, _fields_, _options_).
         1. Perform ? ToTemporalOverflow(_options_).
-        1. Let _string_ be ? ToString(_item_).
-        1. Let _result_ be ? ParseTemporalDateString(_string_).
+        1. If _item_ is not a String, throw a *TypeError* exception.
+        1. Let _result_ be ? ParseTemporalDateString(_item_).
```

Comment:
In line with the reasoning behind #2478 and [how-we-work](https://github.com/tc39/how-we-work/pull/119/files#diff-b48c7b3eba9c6b2a43ae502ddc9eabd1a9c49a2aa5592c64ea8ef548dc81352aR18-R26), I think this should check _item_ before _options_, here and in ToTemporalDateTime/ToTemporalMonthDay/ToTemporalYearMonth/etc.
```suggestion
        1. If _item_ is not a String, throw a *TypeError* exception.
        1. Let _result_ be ? ParseTemporalDateString(_item_).
        1. Perform ? ToTemporalOverflow(_options_).
```
(but maybe this is better suited for #2519—any thoughts, @ptomato?)

Reply:
I don't have a strong opinion either way.

Reply:
I would prefer for this to be handled in #2519 if that's OK, so we can keep this PR focused solely on primitive coercion. So I added https://github.com/tc39/proposal-temporal/pull/2519#issuecomment-1636974266 and will resolve this thread here in this PR.